File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/api/MetadataAiUseCases.java
package com.cleverferret.v2.feature.ai.api;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.ai.model.AiGenerationResult;

public interface MetadataAiUseCases {
    IntegrationResult<AiGenerationResult> summarizeMetadata(String requester, String metadataText);

    IntegrationResult<AiGenerationResult> recommendFromMetadata(String requester, String metadataText);
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/api/MetadataAiUseCases.kt
package com.cleverferret.v2.feature.ai.api

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.ai.model.AiGenerationResult

interface MetadataAiUseCases {
    fun summarizeMetadata(requester: String, metadataText: String): IntegrationResult<AiGenerationResult>
    fun recommendFromMetadata(requester: String, metadataText: String): IntegrationResult<AiGenerationResult>
}
